### DISCUSSION TITLE
Id not generated on `prisma.[model].create` if schema using `@default(dbgenerated("gen_random_uuid()"))`

### DISCUSSION THREAD

**Discussion by danny98m** [prisma/prisma#21351] [Q&A]
### Bug description

When using the `create` api in prisma on a model that has the id with a `@default(dbgenerated("gen_random_uuid()"))` I get a `Null constraint violation on the fields: (`id`)` error as the id remains undefined. When previously using just the prisma `@default(uuid())` this wasn't a problem but I need the uuid generation to also live in the actual postgres schema so needed to use `dbgenerated("gen_random_uuid()")`. 

So it seems when defining the constraint this way the uuid is not generated when prisma create is invoked on a model. Maybe this is intentional which then apologies for the created issue I'd be interested in how I can call the prisma create api in this scenario without triggering a non null constraint (obviously I want Id to have a non null constraint). The uuid default gen is correctly added to the actual db schema and inserting a record directly using sql will correctly generate the uuid id. 

### How to reproduce

**To Reproduce**
Im using prisma client `4.16.1` and my postgres db is using postgres 14.6

Create a schema with a Postgres db where a model uses the `@default(dbgenerated("gen_random_uuid()"))` for an id field that also has the `@id` and `@postgresql.Uuid` attribute. 

Use the `prisma.[modelname].create()` method expecting a record to be created with a uuid generated from the database. but instead receive an error described above.

### Expected behavior

I expected the create api to still work with `@default(dbgenerated("gen_random_uuid()"))` and correctly generate a uuid for the new record. 

### Prisma information

prisma version: `4.16.1`
postgres version: `14.6`

simplified schema example
```prisma
model Account {
  id                        String                 @id @default(dbgenerated("gen_random_uuid()")) @postgresql.Uuid
  name                  String                 @default("")
}
```

example call to prisma create api
```ts
await this.prismaService.account.create(args)
```

### Environment & setup

- OS: macOS, ubuntu container
- Database: PostgreSQL
- Node.js version: 18.13.0
- Prisma: 4.16.1
- Postgres: 14.6

### Prisma Version

```
4.16.1
```

**Comment by janpio** [comment]
Did `gen_random_uuid()` make it to the database as the default value of your column via `db push` or some `migrate` command?
Prisma is not actually invovled in generating the IDs when you use `dbgenerated`.

**Reply by danny98m** [reply]
yes. in the actual db schema `gen_random_uuid` is present as the default after migration and will work when records are inserted directly with sql or through a db management tool. it only fails when using prisma's create as the id ends up undefined which then makes prisma raise the non null error 
```prisma:error 
[0] [start:dev] Invalid `this.prismaService.account.create()` invocation in
[0] [start:dev] 
[0] [start:dev]   16 
[0] [start:dev]   17 async create(args: Prisma.AccountCreateArgs) {
[0] [start:dev]   18    console.log(args);
[0] [start:dev] → 19    const acc = await this.prismaService.account.create(
[0] [start:dev] Null constraint violation on the fields: (`id`)
```

**Reply by janpio** [reply]
How exactly are you trying to create? What data are you inputting?

The error message `Null constraint violation on the fields: (id)` comes from your database, not Prisma. So somehow it leads to Prisma eplicitly trying to insert `null` for that field.

Can you enable query logging so we can see the actual query that is being executed against your database?

**Reply by danny98m** [reply]
I am trying to create by simply using `prisma.account.create({data: {name: 'some name'})` 

Here is the query that gets executed: 
```
[0] [start:dev] prisma:query SELECT 1
[0] [start:dev] prisma:query BEGIN
[0] [start:dev] prisma:query INSERT INTO "public"."Account" ("name","type","accountManagerName","accountManagerEmail","createdAt") VALUES ($1,$2,$3,$4,$5) RETURNING "public"."Account"."id"
[0] [start:dev] prisma:query ROLLBACK
```

those fields of type, accountmanager stuff, and created at are being added by default values. when using `@default(uuid())` there is no issue and the id is properly generated and the record returned but not the case with `gen_random_uuid`

**Reply by janpio** [reply]
Something is off there, the query does not actually try to write `id`. Are you 100% sure that the function is active on `Account.id` on the database server you are talking to? What you are seeing would be how it would look like if there is no function defined in the database, but Prisma assumes there is and hence does not calculate an ID itself.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
